src/inputs/checkbox-radio.scss

```diff
@@ -30,31 +29,42 @@
 
   .iui-checkbox-checkmark,
   .iui-radio-dot {
-    box-sizing: border-box;
     width: $iui-icons-default;
     height: $iui-icons-default;
     margin: 0;
     display: flex;
+    flex-shrink: 0;
     justify-content: center;
     align-items: center;
-
-    @include themed {
-      border: 1px solid rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
-      background-color: t(iui-color-background-1);
+    position: relative;
+    isolation: isolate;
+
+    &::after {
+      content: '';
+      position: absolute;
+      inset: 0;
```

Comment:
You could inherit `border-color` and `border-radius` in here, so that you can set it on the parent for those different variants. This means you will be able to remove `::after` from the selector everywhere else (without even using CSS variables).

```scss
&::after {
  ...
  border-radius: inherit;
  border-color: inherit;
```

Reply:
I thought that this might work `border: 1px solid inherit;` but it did not so I did:
```scss
&::after {
  …
  border-style: solid;
  border-width: 1px;
  border-color: inherit;
  … 
}
```

The only one I didn't change was this portion because it also handles `background-color`:
```scss
input:disabled ~ .iui-checkbox-checkmark::after,
input:disabled ~ .iui-radio-dot::after {
  cursor: not-allowed;
  @include themed {
    background-color: t(iui-color-background-disabled);
    border-color: t(iui-color-background-disabled);
  }
}
```

Reply:
background-color doesn't affect the border, so is that really an issue? You might be able to remove `::after` there too.

Reply:
If I remove `::after` the disabled inputs get the background-color applied on active state.
![Screen Shot 2022-01-07 at 3 12 57 PM](https://user-images.githubusercontent.com/849817/148601553-60936613-506a-4f0e-8822-10c6aa005be5.png)
![Screen Shot 2022-01-07 at 3 13 10 PM](https://user-images.githubusercontent.com/849817/148601570-9f692356-931a-4768-8308-bde99d0bd552.png)

Reply:
But it looks like if I move the background-color out of `::after` and into the parent it does work.

Reply:
The only area I am struggling to remove `::after` from at this point is lines 38-41 in `checkbox.scss`.  It must have something to do with specificity because if I remove `::after` and apply `!important` it does work.

Reply:
You can artificially inflate specificity by adding `&:checked, &:indeterminate`